src/containers/NodePropertyGroupContainer.js

```diff
@@ -1,14 +1,50 @@
 import React, { Component } from "react";
+import THREE from "../vendor/three";
+import PropTypes from "prop-types";
+import { withEditor } from "./EditorContext";
 import PropertyGroup from "../components/PropertyGroup";
 import InputGroup from "../components/InputGroup";
 import Vector3Input from "../components/Vector3Input";
+import StringInput from "../components/StringInput";
+import SetValueCommand from "../editor/commands/SetValueCommand";
+import SetPositionCommand from "../editor/commands/SetPositionCommand";
 
-export default class NodePropertyGroupContainer extends Component {
+class NodePropertyGroupContainer extends Component {
+  static propTypes = {
+    editor: PropTypes.object
+  };
+  constructor(props) {
+    super(props);
+    this.props.editor.signals.objectSelected.add(this.updateNode);
+    this.props.editor.signals.objectChanged.add(this.updateStateFromNode);
+    this.state = { node: {} };
+  }
+  updateNode = node => {
+    this.setState({ node }, () => this.updateStateFromNode(node));
+  };
+  updateStateFromNode = node => {
+    // Ignore objectChanged signals if they are for another node.
+    if (!this.state.node || this.state.node !== node) return;
+    this.setState({
+      name: node.name
+    });
+  };
+  updateName = e => {
+    if (!this.state.node) return;
+    this.props.editor.execute(new SetValueCommand(this.state.node, "name", e.target.value));
+  };
+  updatePosition = newPosition => {
+    if (!this.state.node) return;
+    this.props.editor.execute(new SetPositionCommand(this.state.node, new THREE.Vector3().copy(newPosition)));
```

Comment:
Dunno if it's a bad idea to require and use `THREE` here, but `SetPositionCommand` expects a Vector3.

Reply:
It's probably fine to include `THREE` in the container. I'd try to avoid it any further down the hierarchy.

Reply:
It should be safe to store and set a local THREE.Vector3 every time you update the position. `SetPositionCommand` already clones the new position: https://github.com/MozillaReality/hubs-editor/blob/master/src/editor/commands/SetPositionCommand.js#L28